### DISCUSSION TITLE
Updating meep from source

### DISCUSSION THREAD

**Discussion by kalosu** [NanoComp/meep#2401] [Q&A] [answered]
Hello there meep community,

I am currently using meep after building it from source by following the instructions given in Meep's documentation.
I would like to update my meep by pulling the newest source code with all of the latest merged new changes but I am not really sure if I can just re do the same thing that I did for building it the first time. If I do this, will my meep python modules be automatically updated once I call the functions after re-building from the source code?

Again, thanks for your feedback.

**Comment by smartalecH** [comment, marked as answer]
If you've already built once, then updating your build is rather easy. 

Update your source repo:
```
git pull
```

Remake your sources (in parallel)
```
make -j
```

And reinstall
```
sudo make install
```

No need to reconfigure or rerun `autogen.sh` etc. 

It will overwrite your existing installed files. So long as your `PYTHONPATH` is still set, you should be good to go.

**Reply by kalosu** [reply]
Great, it worked.
Thanks!
